### PR TITLE
2089052 add food allergies

### PR DESCRIPTION
**Pull request by jlandwehr-avarteq**
Created allergies that are associated with users as a N to M relationship. Styled all necessary views and created a link in the header. Furthermore, a lot of refactoring was done. There is now a user controller and user views instead of an user profile. A lot of styling adjustments were also done (almost all views needed to be reworked). The routes were cleaned up, too.

@arkanoryn

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    @user = User.find(params[:id])
+  end
+
+  def edit
+    @user = current_user
+  end
+
+  def update
+    @user = current_user
+    set_allergies_from_params
```
Comment (arkanoryn) [inline app/controllers/users_controller.rb]:
```
@user.allergies = get_allergies_from_params
```

File: app/models/user.rb [hunk 2]
```diff
@@ -18,4 +21,5 @@ class User < ActiveRecord::Base
   validates :first_name, presence: true
   validates :last_name, presence: true
   validates_uniqueness_of :username
+
```
Comment (arkanoryn) [inline app/models/user.rb:23]: space?

File: app/views/allergies/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<div class="container">
+  <div class="row">
+    <div class="col-xs-10 col-xs-offset-1 col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3 col-lg-6 col-lg-offset-3">
+      <%= form_for(@allergy) do |f| %>
+        <div class="form-group">
+          <div class="field">
+            <%= f.label :name %><br>
+            <%= f.text_field :name, class: 'form-control' %>
+          </div>
+          <div class="form-group text-center">
```
Comment (arkanoryn) [inline app/views/allergies/_form.html.erb]: buttons should not be on a form-group

File: app/views/allergies/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<div class="container">
+  <div class="page-header">
+    <div class="row">
+      <h1 class="text-center"><%= t('allergies.allergies_header') %></h1>
+    </div>
+  </div>
+  <div class="row">
+    <div class="col-xs-10 col-xs-offset-1 col-sm-8 col-sm-offset-2 col-md-6 col-md-offset-3 col-lg-6 col-lg-offset-3">
+  <table class="table table-striped">
```
Comment (arkanoryn) [inline app/views/allergies/index.html.erb]: tabulation error

File: app/views/allergies/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    <tbody>
+      <% @allergies.each do |allergy| %>
+        <tr>
+          <td class="col-xs-6 col-sm-6 col-md-4 col-lg-4"><%= allergy.name %></td>
+          <td class="col-xs-2 col-sm-1 col-md-1 col-lg-1"><%= render_allergy_edit_button(allergy) %></td>
+          <td class="col-xs-2 col-sm-1 col-md-1 col-lg-1"><%= render_allergy_destroy_button(allergy) %></td>
+        </tr>
+      <% end %>
+    </tbody>
+  </table>
```
Comment (arkanoryn) [inline app/views/allergies/index.html.erb]: leave some space between rows for better reading experience.

File: app/views/allergies/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  <div class="row">
+    <p class="text-center"><%= t('allergies.add') %></p>
+    <div class="text-center">
+    <%= link_to t('forms.buttons.new'), new_allergy_path, class: 'btn btn-success' %>
```
Comment (arkanoryn) [inline app/views/allergies/index.html.erb]: tab error

**Review by arkanoryn — APPROVED** [review]
can be improve with changes.

**Comment by jlandwehr-avarteq** [comment]
Done.

**Pull request merged**